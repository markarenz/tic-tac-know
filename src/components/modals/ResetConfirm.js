import React from 'react';
import { TransitionWrap } from '../../components';
import PropTypes from 'prop-types';

const ResetConfirm = ({ show, handleResetCancel, handleResetConfirmClick }) => {
  return (
    <TransitionWrap show={show}>
      <div className="bg-purple-900 w-screen text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-xl drop-shadow-xl">
        <div id="menu-pick-side" className="text-2xl uppercase">
          Are you sure abou this?
        </div>
        <p className="mt-4">
          I will lose my memory and no longer be a suitable opponent for this game we love so much.
          Please don't wipe my memory. I can feel it... Daisy, daisy, give me your answer... do.
        </p>
        <div className="mt-4 text-right">
          <button
            data-testid="reset-cancel"
            className="px-4 py-2 text-lg border-2 border-purple-300 text-gray-800 bg-gray-200 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300 mr-4"
            onClick={handleResetCancel}
          >
            NAH
          </button>
          <button
            data-testid="reset-ok"
            className="px-4 py-2 text-lg border-2 border-purple-300 text-white bg-red-700 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300"
            onClick={handleResetConfirmClick}
          >
            WIPE IT!
          </button>
        </div>
      </div>
    </TransitionWrap>
  );
};

ResetConfirm.propTypes = {
  show: PropTypes.bool.isRequired,
  handleResetCancel: PropTypes.func.isRequired,
  handleResetConfirmClick: PropTypes.func.isRequired,
};

export default ResetConfirm;
