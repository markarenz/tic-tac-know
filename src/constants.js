// STRINGS
const PLAYER = 'player';
const CPU = 'cpu';
const X = 'x';
const O = 'o';

const AI_LEVELS = {
  AI_LEVEL_RND: 'ai_level_rnd', // Always random
  AI_LEVEL_KI: 'ai_level_ki', // Killer instinct, random until a winning move is found
  AI_LEVEL_HAH: 'ai_level_hah', // Half and half, sometimes SMRT and sometimes KI
  AI_LEVEL_SMRT: 'ai_level_smrt', // Use past games to learn how to win or draw
};

const symbols = [X, O];
const sides = [PLAYER, CPU];
const coinSides = ['heads', 'tails'];
const initOpenCells = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const winningPatterns = [
  [1, 2, 3], // TOP ROW
  [4, 5, 6], // MIDDLE ROW
  [7, 8, 9], // BOTTOM ROW
  [1, 4, 7], // LEFT COL
  [2, 5, 8], // MIDDLE COL
  [3, 6, 9], // RIGHT COL
  [1, 5, 9], // TL-BR DIAG
  [3, 5, 7], // TR-BL DIAG
];

const LOCAL_STORAGE_KEYS = {
  HISTORY: 'tic-tac-know-history',
  WINS_LOSSES: 'tic-tac-know-wins-losses',
};

const whoseTurnMessages = {
  player: 'Your turn',
  cpu: 'CPU Turn',
};
const gameOverMessages = {
  player: 'YOU WIN!',
  cpu: 'CPU WINS',
  draw: 'The Game is a draw.',
};

const APP_STATES = {
  GAME: 'game',
  MENU: 'menu',
};

const MENU_STATUSES = {
  PICK_SIDE: 'pick-side',
  PICK_AI: 'pick-ai',
  START: 'start',
  COMPLETE: 'complete',
};

const sideMessages = {
  x: 'You are playing as X. The CPU is playing as O.',
  o: 'You are playing as O. The CPU is playing as X.',
};
const aiLevelLabels = {
  ai_level_rnd: 'Herp Derp',
  ai_level_ki: 'Killer Instinct',
  ai_level_hah: 'Half & Half',
  ai_level_smrt: 'Mastermind',
};
const aiLevelMessages = {
  ai_level_rnd:
    'Your oppnent is, to put it bluntly, not terribly bright, making moves (and inapropriate noises) at random.',
  ai_level_ki:
    'Your opponent knows enough to be annoyingly good, but just like all your friends from college, it cannot plan ahead.',
  ai_level_hah:
    'Your opponent is playing with one chip tied behind its back, using a mix of strategies, sometimes smart and sometimes not.',
  ai_level_smrt:
    "Your opponent is smarter than you are, more disciplined and, let's face it, better looking. You don't stand a chance.",
};

//
export {
  symbols,
  sides,
  coinSides,
  PLAYER,
  CPU,
  X,
  O,
  whoseTurnMessages,
  gameOverMessages,
  sideMessages,
  aiLevelMessages,
  aiLevelLabels,
  initOpenCells,
  winningPatterns,
  AI_LEVELS,
  LOCAL_STORAGE_KEYS,
  APP_STATES,
  MENU_STATUSES,
};
