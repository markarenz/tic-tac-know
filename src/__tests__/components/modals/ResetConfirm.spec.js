import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import ResetConfirm from '../../../components/modals/ResetConfirm';
import { AI_LEVELS, X, O } from '../../../constants';

const mocks = {
  show: true,
  playerSide: X,
  handleResetCancel: jest.fn(),
  handleResetConfirmClick: jest.fn(),
};
describe('ResetConfirm - X', () => {
  it('renders component - CANCEL Click', async () => {
    const container = render(
      <ResetConfirm
        show={mocks.show}
        handleResetCancel={mocks.handleResetCancel}
        handleResetConfirmClick={mocks.handleResetConfirmClick}
      />
    );
    await act(async () => {
      fireEvent(
        screen.getByTestId('reset-cancel'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
  it('renders component - OK Click', async () => {
    const container = render(
      <ResetConfirm
        show={mocks.show}
        handleResetCancel={mocks.handleResetCancel}
        handleResetConfirmClick={mocks.handleResetConfirmClick}
      />
    );
    await act(async () => {
      fireEvent(
        screen.getByTestId('reset-ok'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
