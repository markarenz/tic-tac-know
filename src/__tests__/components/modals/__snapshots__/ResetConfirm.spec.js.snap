// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetConfirm - X renders component - CANCEL Click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="absolute left-0 top-0 w-full h-full flex items-center justify-center"
        data-testid="transition-wrap"
      >
        <div
          class="bg-purple-900 w-screen text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-xl drop-shadow-xl"
        >
          <div
            class="text-2xl uppercase"
            id="menu-pick-side"
          >
            Are you sure abou this?
          </div>
          <p
            class="mt-4"
          >
            I will lose my memory and no longer be a suitable opponent for this game we love so much. Please don't wipe my memory. I can feel it... Daisy, daisy, give me your answer... do.
          </p>
          <div
            class="mt-4 text-right"
          >
            <button
              class="px-4 py-2 text-lg border-2 border-purple-300 text-gray-800 bg-gray-200 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300 mr-4"
              data-testid="reset-cancel"
            >
              NAH
            </button>
            <button
              class="px-4 py-2 text-lg border-2 border-purple-300 text-white bg-red-700 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300"
              data-testid="reset-ok"
            >
              WIPE IT!
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="absolute left-0 top-0 w-full h-full flex items-center justify-center"
      data-testid="transition-wrap"
    >
      <div
        class="bg-purple-900 w-screen text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-xl drop-shadow-xl"
      >
        <div
          class="text-2xl uppercase"
          id="menu-pick-side"
        >
          Are you sure abou this?
        </div>
        <p
          class="mt-4"
        >
          I will lose my memory and no longer be a suitable opponent for this game we love so much. Please don't wipe my memory. I can feel it... Daisy, daisy, give me your answer... do.
        </p>
        <div
          class="mt-4 text-right"
        >
          <button
            class="px-4 py-2 text-lg border-2 border-purple-300 text-gray-800 bg-gray-200 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300 mr-4"
            data-testid="reset-cancel"
          >
            NAH
          </button>
          <button
            class="px-4 py-2 text-lg border-2 border-purple-300 text-white bg-red-700 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300"
            data-testid="reset-ok"
          >
            WIPE IT!
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ResetConfirm - X renders component - OK Click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="absolute left-0 top-0 w-full h-full flex items-center justify-center"
        data-testid="transition-wrap"
      >
        <div
          class="bg-purple-900 w-screen text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-xl drop-shadow-xl"
        >
          <div
            class="text-2xl uppercase"
            id="menu-pick-side"
          >
            Are you sure abou this?
          </div>
          <p
            class="mt-4"
          >
            I will lose my memory and no longer be a suitable opponent for this game we love so much. Please don't wipe my memory. I can feel it... Daisy, daisy, give me your answer... do.
          </p>
          <div
            class="mt-4 text-right"
          >
            <button
              class="px-4 py-2 text-lg border-2 border-purple-300 text-gray-800 bg-gray-200 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300 mr-4"
              data-testid="reset-cancel"
            >
              NAH
            </button>
            <button
              class="px-4 py-2 text-lg border-2 border-purple-300 text-white bg-red-700 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300"
              data-testid="reset-ok"
            >
              WIPE IT!
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="absolute left-0 top-0 w-full h-full flex items-center justify-center"
      data-testid="transition-wrap"
    >
      <div
        class="bg-purple-900 w-screen text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-xl drop-shadow-xl"
      >
        <div
          class="text-2xl uppercase"
          id="menu-pick-side"
        >
          Are you sure abou this?
        </div>
        <p
          class="mt-4"
        >
          I will lose my memory and no longer be a suitable opponent for this game we love so much. Please don't wipe my memory. I can feel it... Daisy, daisy, give me your answer... do.
        </p>
        <div
          class="mt-4 text-right"
        >
          <button
            class="px-4 py-2 text-lg border-2 border-purple-300 text-gray-800 bg-gray-200 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300 mr-4"
            data-testid="reset-cancel"
          >
            NAH
          </button>
          <button
            class="px-4 py-2 text-lg border-2 border-purple-300 text-white bg-red-700 rounded-lg scale-100 hover:scale-[1.2] transition-scale font-bold duration-300"
            data-testid="reset-ok"
          >
            WIPE IT!
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
